### PR TITLE
Initialize requested endpoints for the thinthread demo

### PR DESCRIPTION
**Pull request by fivegrant**
Endpoints are added for
- [x] #22 and #20
  - [x] Tests implemented
- [x] #24
  - [x] Tests implemented
- [x] #19
  - [x] Tests implemented  
- [x] #18
  - [x] Tests implemented
  
  The endpoints work but they are definitely a WIP because the needs for the thinthread and the ASKEM project will change.

### CODE REVIEW

**Comment by fivegrant** [comment]
NOTE: Delete is not implemented for these objects because we don't want to mess with the provenance graph. We can update the project to hide objects from the user.

**Comment by brandomr** [comment]
@fivegrant we should indicate in the README how to navigate to the swagger UI (e.g. `localhost:8001`)

File: tds/autogen/schema.py [hunk 1]
```diff
@@ -171,7 +111,6 @@ class Dataset(BaseModel):
     quality: Optional[str]
     temporal_resolution: Optional[str]
     geospatial_resolution: Optional[str]
-    annotations: Optional[Json]
```
Comment (Sorrento110) [inline tds/autogen/schema.py:174]: Data-annotation currently depends on this field to work.

File: tds/autogen/orm.py [hunk 2]
```diff
@@ -189,7 +121,6 @@ class Dataset(Base):
     quality = sa.Column(sa.Text())
     temporal_resolution = sa.Column(sa.String())
     geospatial_resolution = sa.Column(sa.String())
-    annotations = sa.Column(JSON())
```
Comment (Sorrento110) [inline tds/autogen/orm.py:192]: data-annotation depends on this field

**Review by Sorrento110 — CHANGES_REQUESTED** [review]
One breaking change, there are some slight changes data-annotations will have to go through too to accommodate more of these changes but that's all good. The new stuff you implemented is good.

**Review by Sorrento110 — APPROVED** [review]

**Comment by fivegrant** [comment]
Thanks @brandomr , added that to the documentation

**Pull request merged**